modules/transports/optional/fix/src/test/java/org/apache/synapse/transport/fix/FIXTransportListenerTest.java

```diff
@@ -0,0 +1,50 @@
+package org.apache.synapse.transport.fix;
+
+import junit.framework.TestCase;
+import org.apache.axis2.context.ConfigurationContext;
+import org.apache.axis2.description.AxisService;
+import org.apache.axis2.description.Parameter;
+import org.apache.axis2.description.TransportInDescription;
+import org.apache.axis2.description.TransportOutDescription;
+import org.apache.axis2.engine.AxisConfiguration;
+import org.apache.axis2.transport.testkit.axis2.TransportDescriptionFactory;
+import org.junit.Assert;
+import org.junit.Test;
+
+public class FIXTransportListenerTest extends TestCase {
+    @Test
+    public void testFIXTransportListenerInit() throws Exception {
+        AxisService axisService = new AxisService("testFIXService");
+        axisService
+                .addParameter(new Parameter(FIXConstants.FIX_ACCEPTOR_CONFIG_URL_PARAM, "/sample/path/Mock.cfg"));
+        axisService.addParameter(
+                new Parameter(FIXConstants.FIX_INITIATOR_CONFIG_URL_PARAM, "/sample/path/Mock2.cfg"));
+
+        ConfigurationContext cfgCtx = new ConfigurationContext(new AxisConfiguration());
+        TransportDescriptionFactory transportDescriptionFactory = new TransportDescriptionFactory() {
+            @Override
+            public TransportOutDescription createTransportOutDescription() throws Exception {
+                TransportOutDescription trpOutDesc = new TransportOutDescription("fix");
+                trpOutDesc.setSender(new FIXTransportSender());
+                return trpOutDesc;
+            }
+
+            @Override
+            public TransportInDescription createTransportInDescription() throws Exception {
+                TransportInDescription trpInDesc = new TransportInDescription("fix");
+                trpInDesc.setReceiver(new FIXTransportListener());
+                return trpInDesc;
+            }
+        };
+        TransportInDescription transportInDescription = null;
+        try {
+            transportInDescription = transportDescriptionFactory.createTransportInDescription();
+        } catch (Exception e) {
+            e.printStackTrace();
```

Comment:
Shall we have logs instead of printStackTrace

Reply:
done